pls_balances/src/generators.py

```diff
@@ -222,20 +238,34 @@ def generate_balanced_block_table(reps, n_species_class1, n_species_class2,
     """
     data = []
     metadata = []
-    for _ in range(reps):
-        data.append([1]*(n_species_class1+n_species_class2+n_species_shared))
-        metadata += [0]
 
-    for _ in range(reps):
-        data.append(
-            [1/effect_size]*n_species_class1 +
-            [1]*(n_species_shared) +
-            [effect_size]*n_species_class2)
-        metadata += [1]
+    n_species = n_species_class1 + n_species_class2 + n_species_shared
+
+    if template is None:
+      for _ in range(reps):
+          data.append([1]*(n_species))
+          metadata += [0]
+
+      for _ in range(reps):
+          data.append(
+              [1/effect_size]*n_species_class1 +
+              [1]*(n_species_shared) +
+              [effect_size]*n_species_class2)
+          metadata += [1]
+    else:
```

Comment:
can we add `template = closure(template)` just to make sure that the template proportions do add up to 1?